### PR TITLE
[JBIDE-25047] correct width of the "Discover" button

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@robstryker plz review

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void widgetSelected(SelectionEvent e) {
 					}
 				});
 				GridDataFactory.fillDefaults()
-				.align(SWT.FILL, SWT.CENTER).grab(true, false).span(1, 1)
+				.align(SWT.FILL, SWT.CENTER).hint(100, SWT.DEFAULT)
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: Why not using org.jboss.tools.openshift.internal.common.ui.utils.UIUtils.setDefaultButtonWidth ?

Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: UIUtils.setDefaultButtonWidth is as effective while being simpler with the same effect.

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
huston, we have a problem. Jenkins builds as in PR-checks are failing because of internal jenkins problems:
```
An exception occurred while writing to the platform log:
java.io.IOException: No space left on device
	at java.io.FileOutputStream.writeBytes(Native Method)
	at java.io.FileOutputStream.write(FileOutputStream.java:326)
	at sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)
	at sun.nio.cs.StreamEncoder.implFlushBuffer(StreamEncoder.java:291)
	at sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:295)
	at sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:141)
	at java.io.OutputStreamWriter.flush(OutputStreamWriter.java:229)
	at java.io.BufferedWriter.flush(BufferedWriter.java:254)
	at org.eclipse.osgi.internal.log.EquinoxLogWriter.log(EquinoxLogWriter.java:287)
	at org.eclipse.osgi.internal.log.EquinoxLogWriter.logged(EquinoxLogWriter.java:706)
	at org.eclipse.osgi.internal.log.ExtendedLogReaderServiceFactory.safeLogged(ExtendedLogReaderServiceFactory.java:88)
	at org.eclipse.osgi.internal.log.ExtendedLogReaderServiceFactory.logPrivileged(ExtendedLogReaderServiceFactory.java:217)
	at org.eclipse.osgi.internal.log.ExtendedLogReaderServiceFactory.log(ExtendedLogReaderServiceFactory.java:189)
	at org.eclipse.osgi.internal.log.ExtendedLogServiceFactory.log(ExtendedLogServiceFactory.java:65)
	at org.eclipse.osgi.internal.log.ExtendedLogServiceImpl.log(ExtendedLogServiceImpl.java:87)
	at org.eclipse.osgi.internal.log.LoggerImpl.log(LoggerImpl.java:54)
	at org.eclipse.osgi.internal.log.EquinoxLogFactory$1.log(EquinoxLogFactory.java:52)
	at org.eclipse.equinox.internal.p2.core.helpers.LogHelper.log(LogHelper.java:25)
	at org.eclipse.equinox.internal.p2.engine.ProfilePreferences$SaveJob.run(ProfilePreferences.java:59)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Logging to the console instead.

!ENTRY org.eclipse.equinox.p2.engine 2 0 2017-09-19 16:23:08.900
!MESSAGE Exception saving profile preferences
!STACK 0
org.osgi.service.prefs.BackingStoreException: Exception saving preferences to: /tmp/tycho_reactor_agent5189573774900992999/org.eclipse.equinox.p2.engine/.settings/org.eclipse.equinox.p2.artifact.repository.prefs.
	at org.eclipse.core.internal.preferences.EclipsePreferences.write(EclipsePreferences.java:285)
	at org.eclipse.core.internal.preferences.EclipsePreferences.save(EclipsePreferences.java:1076)
	at org.eclipse.equinox.internal.p2.engine.ProfilePreferences.doSave(ProfilePreferences.java:129)
	at org.eclipse.equinox.internal.p2.engine.ProfilePreferences$SaveJob.run(ProfilePreferences.java:49)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.io.FileNotFoundException: /tmp/tycho_reactor_agent5189573774900992999/org.eclipse.equinox.p2.engine/.settings/org.eclipse.equinox.p2.artifact.repository.prefs.bak (No space left on device)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at org.eclipse.core.internal.preferences.SafeFileOutputStream.<init>(SafeFileOutputStream.java:54)
	at org.eclipse.core.internal.preferences.EclipsePreferences.write(EclipsePreferences.java:279)
	... 4 more
```
@nickboldt anything we can do about that?

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
testPR

**Comment by nickboldt** [comment]
To avoid seeing this question every other day, I've started this FAQ:

https://github.com/jbosstools/jbosstools-devdoc/blob/master/building/jenkins_faq.adoc

Good luck!

**Comment by adietish** [comment]
@nickboldt thx, retriggering in jenkins helped. Now we're back to a sonar problem. I'd love to add the -X flag to the maven cmd (to get to know where sonar is NPE'ing exactly) that's run but somehow fail to find it. Have a pointer for me please?

**Comment by nickboldt** [comment]
Easy enough -- just add the flag to the job config.

https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/jbosstools-openshift-Pull-Request/jobConfigHistory/showDiffFiles?timestamp1=2017-07-11_14-43-42&timestamp2=2017-09-20_21-11-46

**Comment by adietish** [comment]
I reported the NPE that we have in Sonar to the mailing-list: https://groups.google.com/forum/?hl=en#!topic/sonarqube/t7-A4M_Ohd0

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
so it turns out that the NPE that we see in sonar definitely is caused by us using org.apache.commons.lang. My latest PR, which - out of curiosity - removes usage of org.apache.commons.lang now does not fail with an NPE in sonar any more (it fails on inclusion/exclusion filters overlap though :( ). 
The issue is filed in sonar (referring to 2.1 but it obviously also occurrs with 2.6 that we're using): https://jira.sonarsource.com/browse/SONARJAVA-2461
Sonar 4.14 apparently fixes it.

**Comment by nickboldt** [comment]
So you want to request that sonar 4.14 be added into Jenkins? If so please open a JBIDE jira and assign to Pavol Srna. I think that's the right place for such a request... if not then it's a PnT JIRA for something upstream in the openstack configs.

**Comment by adietish** [comment]
@nickboldt we're running Sonar in their public instance since we went tired of waiting for our internal infrastructure to become ready. Sonarsource will upgrade their instance to 4.14 in the beginning of next week they say

**Comment by adietish** [comment]
how frustrating: 
I got rid of the NPE by removing references to org.commons.lang. 
Then I hit sonar complaining that a reddeer suite could not be indexed twice, which i fixed by removing the sonar inclusion instruction in pom that added redder classes to the source (and not tests): https://github.com/jbosstools/jbosstools-openshift/pull/1568/commits/680eb1f9b464d641e4bd021a8df6941f01efa5e3
Once this was fixed we're now back with the NPE, but in a different class now.

So there's nothing we can do to have a functional sonar before next week. 

@jeffmaury @bdshadow the inclusion/exclusion instructions for sonar in our current root pom result in the following, that's visible in PR build https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/jbosstools-openshift-Pull-Request/1017/console:
```
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.3.0.603:sonar (default-cli) on project openshift: File tests/org.jboss.tools.openshift.cdk.server.test/src/org/jboss/tools/openshift/cdk/server/test/AllTestsSuite.java can't be indexed twice. Please check that inclusion/exclusion patterns produce disjoint sets for main and test files -> [Help 1]
```
Commenting https://github.com/jbosstools/jbosstools-openshift/pull/1568/commits/680eb1f9b464d641e4bd021a8df6941f01efa5e3
fixes that. Afaics uncommenting treats the reddeer classes as test classes, which I think is correct. Thoughts?

**Comment by bdshadow** [comment]
testPR

**Comment by bdshadow** [comment]
@adietish and me discussed the current sonar NPE problem. According to the sonar https://jira.sonarsource.com/browse/SONARJAVA-2461 the issue will be fixed very soon, even by the end of our sprint. So by now i temporary disabled sonar check in the Jenkins pr check job

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
I now removed all commits that tried to fix the sonar issues, leaving this PR to what it is really all about: fixing button widths for the connection wizard page.
Will merge it once the build is green.

**Comment by adietish** [comment]
testPR

**Pull request merged**